btconfig/utils/dataloader/ib.py

```diff
@@ -163,8 +168,26 @@ def _processRequests(self):
                 whatToShow=str(request['what']),
                 useRTH=int(request['useRTH']),
                 formatDate=1)
+
             while r in self._requests and self.engine.isConnected():
+
+                if request_processed == r:
+                    pass
+                else:  
+                    if i == 1:
+                        start_time = timer()
+                        print(f'Processing {i} / {num_requests} Requests')
+                        i += 1
+                    elif i % 10 == 0:
+                        end_time = timer()
+                        print(f'Processing {i} / {num_requests} Requests. Time lapsed {timedelta(seconds=end_time - start_time)}')
+                        i += 1
+                    else:
+                        print(f'Processing {i} / {num_requests} Requests')
+                        i += 1
+                    request_processed = r
                 sleep(.5)
+
             if not self.engine.isConnected():
                 break
 
```

Comment:
add 
if i % 10 == 0:
    # current time - start time = elapsed time
    self.log(f'log text', logging.DEBUG)
    # update start time to current time
i += 1